RadixWallet/Features/SecurityCenterFeature/SecurityFactors/FactorSourcesList.swift

```diff
@@ -0,0 +1,269 @@
+			return rowsEffect(state: state)
+
+		case let .setEntities(entities):
+			state.entities = entities
+			return rowsEffect(state: state)
+
+		case let .setRows(rows):
+			state.rows = rows
+			return .none
+		}
+	}
+
+	func reduce(into state: inout State, presentedAction: Destination.Action) -> Effect<Action> {
+		switch presentedAction {
+		case .displayMnemonic(.delegate), .enterMnemonic(.delegate), .addMnemonic(.delegate):
+			// We don't care about which delegate action was executed, since any corresponding
+			// updates to the warnings will be handled by securityProblemsEffect.
+			// We just need to dismiss the destination.
+			state.destination = nil
+			return .none
+
+		default:
+			return .none
+		}
+	}
+}
+
+private extension FactorSourcesList {
+	func securityProblemsEffect() -> Effect<Action> {
+		.run { send in
+			for try await problems in await securityCenterClient.problems(.securityFactors) {
+				guard !Task.isCancelled else { return }
+				await send(.internal(.setSecurityProblems(problems)))
+			}
+		} catch: { error, _ in
+			errorQueue.schedule(error)
+		}
+	}
+
+	func entitiesEffect(state: State) -> Effect<Action> {
+		.run { send in
+			let result = try await factorSourcesClient.entitiesLinkedToFactorSourceKind(kind: state.kind)
+			await send(.internal(.setEntities(result)))
+		} catch: { error, _ in
+			errorQueue.schedule(error)
+		}
+	}
+
+	func rowsEffect(state: State) -> Effect<Action> {
+		guard let problems = state.problems, let entities = state.entities else {
+			return .none
+		}
+		return .run { send in
+			let rows = entities.map { entity in
+				let accounts = entity.accounts + entity.hiddenAccounts
+				let personas = entity.personas
+				let status: State.Status = if problems.hasProblem9(accounts: accounts, personas: personas) {
+					.lostFactorSource
+				} else if problems.hasProblem3(accounts: accounts, personas: personas) {
+					.seedPhraseNotRecoverable
+				} else if entity.integrity.isMnemonicMarkedAsBackedUp {
+					.seedPhraseWrittenDown
+				} else {
+					// A way to reproduce this, is to restore a Wallet from a Profile without entering its seed phrase (so it creates a new one)
+					// and do not write down the new seed phrase nor create an entity. In such case, we don't want to show problem3 because the
+					// non-backed up factor source didn't create any entity. Yet, we don't want to show the success checkmark indicating the factor
+					// source was backed up.
+					.notBackedUp
+				}
+				return State.Row(
+					integrity: entity.integrity,
+					linkedEntities: entity.linkedEntities,
+					status: status
+				)
+			}
+			await send(.internal(.setRows(rows)))
+		}
+	}
+}
+
+// MARK: - DeviceFactorSourcesList.State.Row
+extension FactorSourcesList.State {
+	struct Row: Sendable, Hashable {
+		let integrity: FactorSourceIntegrity
+		let linkedEntities: FactorSourceCardDataSource.LinkedEntities
+		let status: Status
+	}
+
+	enum Status: Sendable, Hashable {
```

Comment:
This `Status` enum ought to be in Sargon?

Reply:
mm this one is debatable, could be worth moving to Sargon when working on Security Problems updates mentioned on previous comment.